src/plugins/sensors/sensorfw/sensorfwsensorbase.cpp

```diff
@@ -83,6 +83,10 @@ SensorfwSensorBase::~SensorfwSensorBase()
 
 void SensorfwSensorBase::start()
 {
+    // Application has requested that sensor should be started.
+    // Do it now or/and whenever a sensorfwd start/restart happens.
+    m_attemptRestart = true;
```

Comment:
Here the member variable becomes kind of state of start/stop, i.e. if it's logically on or off. There's also 'running' which gets true if the service start succeeded.

Now, there doesn't seem to be any real need for having these separate. Only reading is connectToSensord() which does attempts reinit if either is true.

-> Could kill the whole m_attemptRestart and just make running true here?

Reply:
Yeah, there is currently two separate things "actually started" = running and "start requested" = m_attemptRestart. The latter having really non-descriptive name especially after changes.

I assumed that "running" would be visible externally but apparently it is not... So, we could squeeze these two booleans in to one. But I would go with "requested" or "startRequested" or stte instead of "running."

Reply:
I'm fine with any names of these. All, including the "running", are better than m_attemptRestart.

Reply:
changed to m_startRequested